internal/controllers/topology/cluster/patches/inline/json_patch_generator_test.go

```diff
@@ -244,13 +244,33 @@ func TestGenerate(t *testing.T) {
 							{
 								Op:   "replace",
 								Path: "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration/controllerManager/extraArgs/cluster-name",
+								ValueFrom: &clusterv1.JSONPatchValue{
```

Comment:
This was a drive-by. I changed this test case to a simple lookup of `builtin.cluster.name`, as it looked like it suits the target path of `cluster-name` more than the `worker-node-azure.json` secret it was before. If I understand this correctly.

Reply:
Hm, I thought that the selector selects the `BootstrapTemplate` within the `MachineDeploymentClass`, which will be e.g. a kubeadmConfig?! This one has `clusterConfiguration`.

Reply:
Yup it has the field but it is not filled out because it would be unused on worker nodes. Only control plane nodes use this part of the configuration

Reply:
I'll update to a slightly more realistic version